src/Simulation/CsharpGeneration/EntryPoint.fs

```diff
@@ -0,0 +1,205 @@
+open System
+open System.IO
+open System.Reflection
+
+
+/// An entry point parameter.
+type private Parameter =
+    { Name : string
+      QsharpType : ResolvedType
+      CsharpTypeName : string
+      Description : string }
+
+/// The namespace in which to put generated code for the entry point.
+let internal generatedNamespace entryPointNamespace = entryPointNamespace + ".__QsEntryPoint__"
+
+/// A public constant field.
+let private constant name typeName value =
+    ``field`` typeName name [``public``; ``const``] (``:=`` value |> Some)
+
+/// A public static property with a getter.
+let private readonlyProperty name typeName value =
+    ``property-arrow_get`` typeName name [``public``; ``static``]
+        ``get`` (``=>`` value)
+
+/// A static class containing constants used by the entry point driver.
+let private constantsClass =
+    ``class`` "Constants" ``<<`` [] ``>>``
+        ``:`` None ``,`` []
+        [``internal``; ``static``]
+        ``{``
+            [readonlyProperty "SimulatorOptions" "System.Collections.Generic.IEnumerable<string>"
+                (``new array`` (Some "") [``literal`` ("--" + fst CommandLineArguments.SimulatorOption)
+                                          ``literal`` ("-" + snd CommandLineArguments.SimulatorOption)])
+             constant "QuantumSimulator" "string" (``literal`` AssemblyConstants.QuantumSimulator)
+             constant "ToffoliSimulator" "string" (``literal`` AssemblyConstants.ToffoliSimulator)
+             constant "ResourcesEstimator" "string" (``literal`` AssemblyConstants.ResourcesEstimator)]
+        ``}``
+
+/// A sequence of all of the named parameters in the argument tuple and their respective C# and Q# types.
+let rec private parameters context doc = function
+    | QsTupleItem variable ->
+        match variable.VariableName, variable.Type.Resolution with
+        | ValidName name, ArrayType itemType ->
+            // Command-line parsing libraries can't convert to IQArray. Use IEnumerable instead.
+            let typeName = sprintf "System.Collections.Generic.IEnumerable<%s>"
+                                   (SimulationCode.roslynTypeName context itemType)
+            Seq.singleton { Name = name.Value
+                            QsharpType = variable.Type
+                            CsharpTypeName = typeName
+                            Description = ParameterDescription doc name.Value }
+        | ValidName name, _ ->
+            Seq.singleton { Name = name.Value
+                            QsharpType = variable.Type
+                            CsharpTypeName = SimulationCode.roslynTypeName context variable.Type
+                            Description = ParameterDescription doc name.Value }
+        | InvalidName, _ -> Seq.empty
+    | QsTuple items -> items |> Seq.map (parameters context doc) |> Seq.concat
+
+/// The custom argument handler for the given Q# type.
+let private argumentHandler =
+    function
+    | UnitType -> Some "UnitArgumentHandler"
```

Comment:
I saw that for entry points that take Unit no argument is required. Nice!

Reply:
Yes, although that is not related to the `UnitArgumentHandler`. Command line options are created for each named argument in the entry point's argument tuple. Since `()` has no named items, there are no options. `UnitArgumentHandler` is for when a named item is of type `Unit`, such as `(foo : Unit)`, and in that case the argument is required. :)

Reply:
I know, I just randomly thought of it when I saw this. :)